reactioncommerce/reaction
# Handle and title updated together

This PR
1. Makes sure when the product is added, the handle created for it is new and unique
2. When a user manually updates the handle, `copy-<'' or some-number>` would be added to it (if required) to keep it unique.

Fix for #2831 #2560 

To test ->
1. run `reaction`
2. Create a product named "Test Product" and publish it.
3. Open the Product settings, the permalink would be `test-product`
4. Create another product and set its `Title` to "Test Product", the permalink this time should be `test-product-copy`
5. Publish the product
6. Create another product with Title "Test Product". It's permalink would be `test-product-copy-1`. Edit it manually to `test-product`. The permalink should change to `test-product-copy-1` automatically.

Suggestion ->
Maybe we can change the `copy` to `item`. So names would be like `test-product`, `test-product-item`, `test-product-item-1`, ...
